File: main/src/main/java/com/sedmelluq/discord/lavaplayer/format/transcoder/AudioChunkEncoder.java
package com.sedmelluq.discord.lavaplayer.format.transcoder;

import java.nio.ShortBuffer;

/**
 * Encodes one chunk of audio from internal PCM format.
 */
public interface AudioChunkEncoder {
  /**
   * @param buffer Input buffer containing the PCM sampels
   * @return Encoded bytes
   */
  byte[] encode(ShortBuffer buffer);

  /**
   * Frees up all held resources.
   */
  void close();
}

File: main/src/main/java/lavaplayer/format/transcoder/AudioChunkEncoder.kt
package lavaplayer.format.transcoder

import java.nio.ByteBuffer
import java.nio.ShortBuffer

/**
 * Encodes one chunk of audio from internal PCM format.
 */
interface AudioChunkEncoder {
    /**
     * @param buffer Input buffer containing the PCM samples.
     * @return Encoded bytes
     */
    fun encode(buffer: ShortBuffer): ByteArray

    /**
     * @param input  Input buffer containing the PCM samples.
     * @param output Output buffer to store the encoded bytes in
     */
    fun encode(input: ShortBuffer, output: ByteBuffer)

    /**
     * Frees up all held resources.
     */
    fun close()
}
